wikimedia/wikipedia-ios
# Talk pages - Populate header with live data (Part 2)

**Phabricator:** 
https://phabricator.wikimedia.org/T310273

### Notes
This is the final portion of the header task. It refactors `RemoteNotificationsProject` for reuse in talk pages, and adjusts our routing to handle non-Wikipedia talk pages natively. Since this touches some deeper shared routing code with the main Wikipedia scheme, I'm going to mark as WIP until 6.9.3 is out.

### Test Steps
This one is a bit difficult to test, since you have to have triggered a talk page notification on a non-Wikipedia project. I have a [sample notification API response](https://github.com/wikimedia/wikipedia-ios/files/9647845/other-project-talk-page-notifs.txt) with a couple of these from my account, which you can try mapping locally via a proxy if interested.

1. Tap a link to a Wikipedia talk page in Notifications Center. Confirm it routes to our new native talk page like before. Confirm language code displays correctly in header like before.
2. (Optional) Tap a link to a non-Wikipedia talk page in Notifications Center. Confirm it routes to our new native talk page. Confirm icon and language code displays correctly in the header. (If mapping locally from linked response above, it will display as empty topics. This is fine for now).

### Screenshots/Videos
![Screen Shot 2022-09-26 at 11 08 05 AM](https://user-images.githubusercontent.com/3620196/192326685-4c80845b-42fb-4d6d-a732-f62a53c00594.png)
![Screen Shot 2022-09-26 at 11 08 09 AM](https://user-images.githubusercontent.com/3620196/192326787-b0822cee-a84f-4d3f-9550-6e1c65852db9.png)
![Screen Shot 2022-09-26 at 11 08 16 AM](https://user-images.githubusercontent.com/3620196/192326769-41875303-4354-4855-b844-a6984d51ccb0.png)
![Screen Shot 2022-09-26 at 11 08 19 AM](https://user-images.githubusercontent.com/3620196/192326722-fb1e8c77-0eaf-4341-869a-a2f7a0b5b598.png)
